File: tests/benchmarks/resolution/fixtures/dynamic-lua/dispatch.lua
-- Phase 6: Lua dynamic dispatch patterns
-- t["method"]() → computed-literal, resolved
-- load(code) → eval kind, flagged
-- t[k]() → computed-key kind, flagged

local function greet(name)
  return "Hello, " .. name
end

-- t["greet"]() — computed-literal with string key; resolves to greet()
local function run_literal_dispatch(t)
  t["greet"]("world")
end

-- load(code) — dynamic code execution; always flagged
local function run_load(code)
  local fn = load(code)
  if fn then fn() end
end

-- t[k]() — variable key; flagged as computed-key
local function run_variable_dispatch(t, k)
  t[k]("world")
end

File: tests/benchmarks/resolution/fixtures/dynamic-elixir/dispatch.ex
defmodule DynamicDispatch do
  # Fixture: Elixir dynamic dispatch via apply/3
  # apply(module, :function, args) → reflection kind
  # apply(module, variable, args) → computed-key

  def greet(name), do: "Hello, #{name}"
  def farewell(name), do: "Goodbye, #{name}"

  # apply(DynamicDispatch, :greet, ["world"]) — reflection kind
  def run_apply_atom do
    apply(DynamicDispatch, :greet, ["world"])
  end

  # apply(module, fn_name, args) — computed-key kind
  def run_apply_variable(fn_name) do
    apply(DynamicDispatch, fn_name, ["world"])
  end
end
